### DISCUSSION TITLE
Initialize a model with terminal state of a solution?

### DISCUSSION THREAD

**Discussion by YiJU-thu** [pybamm-team/PyBaMM#2652] [Q&A]
Hi, I am working on optimal control of energy systems like microgrid or V2G, and I am wondering if it's possible to use PyBaMM as a digital twin to track the battery degradation.

To be specific, my ideal procedures are as follows:

1. My own algorithm will solve a (dis)charging schedule for the battery
2. Then, I want to apply such schedules to a PyBaMM model (this step, I know I can use experiment & simulation)
3. I'll use the terminal state PyBaMM gives to update the parameters in my algorithms, e.g., the battery capacity and the maximum chargingr rate may decay
4. Then, basically, do 1 & 2 again, but I want to **initialize the PyBaMM model with the terminal state of step 2** (i.e., from the last iteration). I get stuck in how to do this - what parameters I need to read from the solution and use them to set the parameter_values - considering my goal stated in step 3?

Thank you in advance for your answer!

**Comment by brosaplanella** [comment]
You can use the `starting_solution` option when you solve, see the Starting solution section on the [long experiment notebook](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/simulating-long-experiments.ipynb). This [notebook](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/initialize-model-with-solution.ipynb) might also be helpful.
